src/AssemblerHttpServer-Alignments.cpp

```diff
@@ -1182,6 +1183,255 @@ void Assembler::computeAllAlignments(
+            computeAllAlignmentsData.downsamplingFactor,
+            computeAllAlignmentsData.bandExtend,
+            html
+    );
+
+    html << "</form>";
+
+    vector<OrientedReadId> sampledReads;
+
+    // If the user input is missing, stop here.
+    if(not samplesIsPresent) {
+        return;
+    }
+
+    // If the user doesn't care about filtering length, sample uniformly
+    if(not minLengthIsPresent and not maxLengthIsPresent) {
+        sampleReads(sampledReads, samples);
+    }
+    // Or else use any provided filters (defaults are used if only one is set)
+    else{
+        sampleReads(sampledReads, samples, minLength, maxLength);
+    }
+
+    // Initialize histograms
+    IterativeHistogram alignedFractionHistogram(0,1,20);
+    IterativeHistogram markerCountHistogram(0,3000,100);
+    IterativeHistogram nAlignmentsHistogram(0,200,20);
+
+    vector<pair<OrientedReadId, AlignmentInfo> > allAlignments;
+    vector<pair<OrientedReadId, AlignmentInfo> > allStoredAlignments;
+
+    for (auto& orientedReadId: sampledReads) {
+        // Vectors to contain markers sorted by kmerId.
+        vector<MarkerWithOrdinal> markers0SortedByKmerId;
+        vector<MarkerWithOrdinal> markers1SortedByKmerId;
+        getMarkersSortedByKmerId(orientedReadId, markers0SortedByKmerId);
+
+        // Compute the alignments in parallel.
+        computeAllAlignmentsData.orientedReadId0 = orientedReadId;
+        const size_t threadCount = std::thread::hardware_concurrency();
+        computeAllAlignmentsData.threadAlignments.resize(threadCount);
+        const size_t batchSize = 1000;
+        setupLoadBalancing(reads.size(), batchSize);
+        const auto t0 = std::chrono::steady_clock::now();
+        runThreads(&Assembler::computeAllAlignmentsThreadFunction, threadCount);
+        const auto t1 = std::chrono::steady_clock::now();
+        html << "<p>Alignment computation using " << threadCount << " threads took " <<
+             1.e-9 * double((std::chrono::duration_cast<std::chrono::nanoseconds>(t1 - t0)).count()) << "s.";
+
+        // Gather the alignments found by each thread.
+        vector<pair<OrientedReadId, AlignmentInfo> > alignments;
+        for (size_t threadId = 0; threadId < threadCount; threadId++) {
+            const vector<pair<OrientedReadId, AlignmentInfo> >& threadAlignments =
+                    computeAllAlignmentsData.threadAlignments[threadId];
+            copy(threadAlignments.begin(), threadAlignments.end(), back_inserter(alignments));
+        }
+        computeAllAlignmentsData.threadAlignments.clear();
+        sort(alignments.begin(), alignments.end(),
+             OrderPairsByFirstOnly<OrientedReadId, AlignmentInfo>());
+
+        // Loop over the STORED alignments that this oriented read is involved in, with the proper orientation.
+        const vector< pair<OrientedReadId, AlignmentInfo> > storedAlignments =
+                findOrientedAlignments(orientedReadId);
```

Comment:
If you only need the number of alignments, you can use `alignmentTable.size(orientedReadId.getValue())` instead.

Reply:
Yea, I also put them all in a vector because for the next step I want to do a table of comparisons for stored vs found alignments. I think the comparison table will help with dead ends (e.g. # of overhangs). Waiting for the next PR if that's alright